include/lang_utils/tuple.h

```diff
@@ -25,52 +26,41 @@ class transform_tuple_type {
                           std::tuple_size<ORIG>::value>>::type;
 };
 
-template <typename ...TUPLES>
-struct tuple_sizes_equal;
+template <size_t N>
+constexpr size_t check_array_equal(size_t const (&arr)[N]) {
+    size_t ret = 0;
+    for (size_t s : arr) {
+        ret = (s == arr[0]) ? s : throw "Sizes mismatch";
```

Comment:
This is a fun trick! If the condition is false, the compilation will fail because `throw` is not allowed in `constexpr` functions. If the condition is true, the `throw` is never evaluated and this whole thing compiles.